src/mmio_types/display_control.rs

```diff
@@ -0,0 +1,34 @@
+use super::*;
+
+#[derive(Debug, Clone, Copy, Default, PartialEq, Eq)]
+#[repr(transparent)]
+pub struct DisplayControl(u16);
+impl DisplayControl {
+  const_new!();
+  bitfield_enum!(u16; 0..=2: DisplayMode, display_mode, with_display_mode, set_display_mode);
+  bitfield_bool!(u16; 4, display_frame1, with_display_frame1, set_display_frame1);
+  bitfield_bool!(u16; 5, hblank_interval_free, with_hblank_interval_free, set_hblank_interval_free);
+  bitfield_bool!(u16; 6, obj_vram_1d, with_obj_vram_1d, set_obj_vram_1d);
+  bitfield_bool!(u16; 7, forced_blank, with_forced_blank, set_forced_blank);
+  bitfield_bool!(u16; 8, display_bg0, with_display_bg0, set_display_bg0);
+  bitfield_bool!(u16; 9, display_bg1, with_display_bg1, set_display_bg1);
+  bitfield_bool!(u16; 10, display_bg2, with_display_bg2, set_display_bg2);
+  bitfield_bool!(u16; 11, display_bg3, with_display_bg3, set_display_bg3);
+  bitfield_bool!(u16; 12, display_obj, with_display_obj, set_display_obj);
+  bitfield_bool!(u16; 13, display_win0, with_display_win0, set_display_win0);
+  bitfield_bool!(u16; 14, display_win1, with_display_win1, set_display_win1);
+  bitfield_bool!(u16; 15, display_obj_win, with_display_obj_win, set_display_obj_win);
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[repr(u16)]
+pub enum DisplayMode {
+  _0 = 0,
+  _1 = 1,
+  _2 = 2,
+  _3 = 3,
+  _4 = 4,
+  _5 = 5,
+  _6 = 6,
```

Comment:
As far as I can tell, this enum is currently part of the public API.

However, display modes 6 and 7 are not valid, and thus shouldn't be used or exposed to the user.

Reply:
On the discord they said that they do draw non-garbage: you get the backdrop color everywhere and objects still draw normal. Which isn't super great but won't crash the device.

i do need docs on this, but since these are being get/set out of the low 3 bits of a u16 I'd rather this type be "complete" (the full 0 to 7 range) even if not all options are useful. the reason is that ever making a rust enum of an invalid bit pattern is UB, so I'd rather have the full range of bit patterns as long as it won't literally crash the GBA.

but yeah I'm not super enthused by this particular bit :/

Reply:
In theory, if all /safe/ accesses to REG_DISPCNT have to go through the enum, removing the 6 and 7 variants would be fine, as any *other* access to REG_DISPCNT which may set these modes would require unsafe, and is thus "not our business"

That's more personal opinion though and less a recommendation of what to do

Reply:
very good point, very good point. I suppose for now we can take off the useless modes. might add them back later.